### DISCUSSION TITLE
How to pick one magnet in one specific cell instead of all of the same type

### DISCUSSION THREAD

**Discussion by yysigari** [atcollab/at#717] [Python AT]
I am trying to change only one of the corrector magnet's kick angle ( corrector number 13 when I use get_cells to pick the correctors) , but when I type this it changes all of the corrector magnet's kick angles. Any idea?  

New_ring. = at.set_value_refpts(Ring, SR[ref_corrector_all][13], 'KickAngle', 10e-6,0, 'copy', True)

**Comment by lcarver** [comment]
Hello, 

Are you defining your lattice in the same script? This is a common problem of python pointers. If you do the following:

```
cor = at.Corrector(args)
ring = []
for i in np.arange(10):
    ring.append(cor)
```
Then every element is a pointer to the same memory. So if you modify one corrector, you modify them all. The correct way to do it would be

```
ring = []
for i in np.arange(10):
    ring.append(at.Corrector(args))
```
Now, every corrector has its own memory and can be modified independently.

Alternatively, you could also do
```
cor = at.Corrector(args)
ring = []
for i in np.arange(10):
    ring.append(cor.copy())
```

Or you could save the lattice to file, then import it. This also breaks the pointers.
